### PR TITLE
[SQL Lab] Allow running multiple statements

### PR DESCRIPTION
**Pull request by mistercrunch**
+ some refactoring to enable this

### CODE REVIEW

**Comment by putop** [comment]
Please stop sending me emails and unsubscribe me

**Comment by mistercrunch** [comment]
@putop you manage your own Github notifications, nothing we can do on this end.

**Comment by mistercrunch** [comment]
@betodealmeida @hughhhh

File: superset/migrations/versions/de021a1ca60d_.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Revision ID: de021a1ca60d
+Revises: ('0b1f1ab473c0', 'cefabc8f7d38')
+Create Date: 2018-12-18 22:45:55.783083
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'de021a1ca60d'
+down_revision = ('0b1f1ab473c0', 'cefabc8f7d38')
```
Comment (betodealmeida) [inline superset/migrations/versions/de021a1ca60d_.py]: Alembic can't merge 3 heads into 1?

File: superset/sql_lab.py [hunk 2]
```diff
@@ -191,78 +169,121 @@ def handle_error(msg):
...
 
-    cdf = dataframe.SupersetDataFrame(data, cursor_description, db_engine_spec)
+    engine = database.get_sqla_engine(
+        schema=query.schema,
+        nullpool=True,
+        user_name=user_name,
+    )
+    # Sharing a single connection and cursor across the
+    # execution of all statements (if many)
+    with closing(engine.raw_connection()) as conn:
```
Comment (betodealmeida) [inline superset/sql_lab.py]: I wish all DB modules had a context manager for connections and cursors...

File: superset/sql_lab.py [hunk 3]
```diff
@@ -191,78 +169,121 @@ def handle_error(msg):
...
-        logging.info('Storing results in results backend, key: {}'.format(key))
-        write_to_results_backend_start = now_as_float()
-        json_payload = json.dumps(
-            payload, default=json_iso_dttm_ser, ignore_nan=True)
-        cache_timeout = database.cache_timeout
-        if cache_timeout is None:
-            cache_timeout = config.get('CACHE_DEFAULT_TIMEOUT', 0)
-        results_backend.set(key, zlib_compress(json_payload), cache_timeout)
+        key = str(uuid.uuid4())
+        logging.info(f'Storing results in results backend, key: {key}')
```
Comment (betodealmeida) [inline superset/sql_lab.py]: Yeah, f strings!

File: superset/views/core.py [hunk 4]
```diff
@@ -2390,11 +2391,11 @@ def results(self, key):
         if not results_backend:
             return json_error_response("Results backend isn't configured")
 
-        read_from_results_backend_start = utils.now_as_float()
+        read_from_results_backend_start = now_as_float()
```
Comment (betodealmeida) [inline superset/views/core.py]: Why not use the context manager here?

**Review by betodealmeida — COMMENTED** [review]
LGTM, just a few comments.

File: superset/migrations/versions/de021a1ca60d_.py (hunk 1 above)
Comment (mistercrunch) [reply superset/migrations/versions/de021a1ca60d_.py]: Oh I'm sure it does, I just rebased 2 times, let me collapse these

**Comment by aboganas** [comment]
I started facing problems with the update statement with this pr
explained here #6706

**Pull request merged**